modules/images/regenerate-existing-images/load.php

```diff
@@ -7,3 +7,79 @@
+ * This taxonomy will be used as a queue system where each term represents a job in the queue.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_background_job_taxonomy() {
+	// Get all registered public post types.
+	$object_types = get_post_types(
+		array(
+			'public' => true,
+		),
+		'names'
+	);
+
+	// Labels for the background job taxonomy.
+	$labels = array(
+		'name'                  => _x( 'Background Jobs', 'taxonomy general name', 'performance-lab' ),
+		'singular_name'         => _x( 'Background Job', 'taxonomy singular name', 'performance-lab' ),
+		'search_items'          => __( 'Search Background Jobs', 'performance-lab' ),
+		'all_items'             => __( 'All Background Jobs', 'performance-lab' ),
+		'parent_item'           => __( 'Parent Background Job', 'performance-lab' ),
+		'parent_item_colon'     => __( 'Parent Background Job:', 'performance-lab' ),
+		'edit_item'             => __( 'Edit Background Job', 'performance-lab' ),
+		'update_item'           => __( 'Update Background Job', 'performance-lab' ),
+		'add_new_item'          => __( 'Add New Background Job', 'performance-lab' ),
+		'new_item_name'         => __( 'New Background Job Name', 'performance-lab' ),
+		'menu_name'             => __( 'Background Jobs', 'performance-lab' ),
+		'view_item'             => __( 'View Background Job', 'performance-lab' ),
+		'not_found'             => __( 'Background Job Not Found', 'performance-lab' ),
+		'no_terms'              => __( ' No Background Jobs', 'performance-lab' ),
+		'items_list'            => __( 'Background Jobs List', 'performance-lab' ),
+		'items_list_navigation' => __( 'Background Jobs List Navigation', 'performance-lab' ),
+		'back_to_items'         => __( 'Back to Background Jobs', 'performance-lab' ),
+	);
+
+	// Taxonomy arguments.
+	$args = array(
+		'hierarchical'       => false, // We do not need child job.
+		'labels'             => $labels,
+		'public'             => false,
+		'show_ui'            => true,
+		'show_in_menu'       => false,
+		'show_in_nav_menus'  => false,
+		'show_in_quick_edit' => false,
+		'show_admin_column'  => false,
+		'show_in_rest'       => false,
+		'query_var'          => false,
+		'capabilities'       => perflab_get_background_job_capabilities(),
+	);
+
+	// Register background_job taxonomy.
+	register_taxonomy( 'background_job', $object_types, $args );
+}
+add_action( 'init', 'perflab_register_background_job_taxonomy', 100 );
+
+/**
+ * Retrieves list of capabilities to manage background jobs.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Map of core term capabilities and the actual capability names to use.
+ */
+function perflab_get_background_job_capabilities() {
+	return array(
+		'manage_terms' => 'manage_jobs', // To manage background_job terms.
+		'edit_terms'   => 'edit_jobs', // To edit background_job terms.
+		'delete_terms' => 'delete_jobs', // To delete background_job terms.
+		'assign_terms' => 'assign_jobs', // To assign background_job terms to posts.
```

Comment:
This is a system taxonomy which means that it will be processed in a background where there is no current user. Thus I am not sure that this capabilities will be really needed and will make any sense at all... 🤔 

cc @felixarntz @adamsilverstein @jjgrainger

Reply:
@eugene-manuilov The intention is to use custom capabilities to hide the admin screen initially but allow developers to access this by adding these capabilities to their own user. The screen is mean for debugging purposes for now and not mean to be available for all admins.

Reply:
The standard taxonomy screen does not make any sense for background jobs even if we hide the second for creating new terms there and leave just the table. We will need to customise it so deeply to provide useful details, so I think it would be better to create a new tools page for it from scratch.

Reply:
I agree that the capabilities still make sense in order to manage the jobs - regardless of whether we implement a new screen or modify the default one.

FWIW, since we're building this for WP core and since it's an advanced utility screen anyway that would be enabled by default, I think we should use the default screen and customize it as needed. This way we reuse the existing infrastructure and keep the work minimal, which given that this screen won't be ever used by most users is reasonable IMO.